src/iiif/ingress_zip_consumer.py

```diff
@@ -27,32 +27,38 @@ class ZipConsumer(BaseConsumer):
-        record = json.loads(raw_data)
+        log.info("Started consume_raw_data")
+        try:
+            record = json.loads(raw_data)
 
-        # Download all the files from the source systems through cantaloupe
-        tmp_folder_path, info_txt_contents, zipjob_uuid = cantaloupe.download_files_for_zip(record)
+            # Download all the files from the source systems through cantaloupe
+            tmp_folder_path, info_txt_contents, zipjob_uuid = cantaloupe.download_files_for_zip(record)
 
-        # Store the info_file_along_with_the_image_files
-        zip_tools.save_file_to_folder(tmp_folder_path, 'report.txt', info_txt_contents)
+            # Store the info_file_along_with_the_image_files
+            zip_tools.save_file_to_folder(tmp_folder_path, 'report.txt', info_txt_contents)
 
-        # Zip all files together
-        zip_file_path = zip_tools.create_local_zip_file(zipjob_uuid, tmp_folder_path)
-        zip_file_name = os.path.basename(zip_file_path)
+            # Zip all files together
+            zip_file_path = zip_tools.create_local_zip_file(zipjob_uuid, tmp_folder_path)
+            zip_file_name = os.path.basename(zip_file_path)
 
-        # Move the file to the object store
-        conn = object_store.get_object_store_connection()
-        object_store.store_object_on_object_store(conn, zip_file_path, zip_file_name)
+            # Move the file to the object store
+            conn = object_store.get_object_store_connection()
+            object_store.store_object_on_object_store(conn, zip_file_path, zip_file_name)
 
-        # Create a temporary url
-        temp_zip_download_url = object_store.create_object_store_temp_url(
-            conn,
-            zip_file_name,
-            expiry_days=settings.TEMP_URL_EXPIRY_DAYS)
+            # Create a temporary url
+            temp_zip_download_url = object_store.create_object_store_temp_url(
+                conn,
+                zip_file_name,
+                expiry_days=settings.TEMP_URL_EXPIRY_DAYS)
 
-        # Send the email
-        email_subject = "Downloadlink Bouw- en omgevingdossiers"
-        email_body = render_to_string('download_zip.html', {'temp_zip_download_url': temp_zip_download_url})
-        mailing.send_email(record['email_address'], email_subject, email_body)
+            # Send the email
+            email_subject = "Downloadlink Bouw- en omgevingdossiers"
+            email_body = render_to_string('download_zip.html', {'temp_zip_download_url': temp_zip_download_url})
+            mailing.send_email(record['email_address'], email_subject, email_body)
 
-        # Cleanup the local zip file and folder with images
-        zip_tools.cleanup_local_files(zip_file_path, tmp_folder_path)
+            # Cleanup the local zip file and folder with images
+            zip_tools.cleanup_local_files(zip_file_path, tmp_folder_path)
+
+        except Exception as e:
+            log.error("ingress_zip_consumer_error:", e)
```

Comment:
Dit zal zelf een exception veroorzaken...

```
Python 3.8.10 (default, Jun  2 2021, 10:49:15)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import logging
>>> log = logging.getLogger('log')
>>> log.error("ingress_zip_consumer_error:", Exception())
--- Logging error ---
Traceback (most recent call last):
  File "/usr/lib/python3.8/logging/__init__.py", line 1085, in emit
    msg = self.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 929, in format
    return fmt.format(record)
  File "/usr/lib/python3.8/logging/__init__.py", line 668, in format
    record.message = record.getMessage()
  File "/usr/lib/python3.8/logging/__init__.py", line 373, in getMessage
    msg = msg % self.args
TypeError: not all arguments converted during string formatting
Call stack:
  File "<stdin>", line 1, in <module>
Message: 'ingress_zip_consumer_error:'
Arguments: (Exception(),)
```

Een betere optie is om [logging.exception](https://docs.python.org/3/library/logging.html#logging.Logger.exception) te gebruiken, waardoor de exception informatie (+traceback) automatisch gelogd wordt...

```
try:
  ...
except:
  log.exception("ingress_zip_consumer_error")
```